native/core/src/execution/planner.rs

```diff
@@ -181,6 +179,61 @@ impl PhysicalPlanner {
         }
     }
 
+    /// get DataFusion PartitionedFiles from a Spark FilePartition
+    fn get_partitioned_files(
+        &self,
+        partition: &SparkFilePartition,
+    ) -> Result<Vec<PartitionedFile>, ExecutionError> {
+        let mut files = Vec::with_capacity(partition.partitioned_file.len());
+        partition.partitioned_file.iter().try_for_each(|file| {
+            assert!(file.start + file.length <= file.file_size);
+
+            let mut partitioned_file = PartitionedFile::new_with_range(
+                String::new(), // Dummy file path.
+                file.file_size as u64,
+                file.start,
+                file.start + file.length,
+            );
+
+            // Spark sends the path over as URL-encoded, parse that first.
+            let url = Url::parse(file.file_path.as_ref()).unwrap();
+            // Convert that to a Path object to use in the PartitionedFile.
+            let path = Path::from_url_path(url.path()).unwrap();
```

Comment:
It would be nice to remove these unwraps and return Err. I know that the goal of this PR is to reorganize the code so we can look at removing unwraps as a separate PR if that is easier.